File: python/22a_1335A_CandiesAndTwoSisters.py
def main():
    t = int(input())
    ans = []
    for _ in range(t):
        temp = int(input())
        if temp % 2 != 0:
            ans.append(temp // 2)
        else:
            ans.append((temp // 2) - 1)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/22a_1335A_CandiesAndTwoSisters.cpp
#include<bits/stdc++.h>

int main(){
    int t, temp;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>temp;
        if(temp % 2 != 0){
            ans.push_back(temp / 2);
        }
        else{
            ans.push_back((temp / 2) - 1);
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}